driver-core/src/main/com/mongodb/internal/connection/ScramShaAuthenticator.java

```diff
@@ -132,7 +140,7 @@ public boolean hasInitialResponse() {
             if (!MessageDigest.isEqual(decodeBase64(map.get("v")), serverSignature)) {
                 throw new SaslException("Server signature was invalid.");
             }
-            return challenge;
+            return new byte[0];
```

Comment:
Once the server signature is validated, we should return an empty byte array according to spec.